### PR TITLE
telemetry: add is_known_ci_env

### PR DESCRIPTION
**Pull request by yuhan**
### Summary & Motivation
explicitly track whether it's a CI env

### How I Tested These Changes

https://buildkite.com/dagster/dagster/builds/45285#0185a23c-4664-49bd-bbeb-f9e59fa2735a/6-958

### CODE REVIEW

**Comment by yuhan** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11624** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11624" target="_blank">Graphite</a> 
    * **PR #11642** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11642" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11642?utm_source=stack-comment).

**Comment by yuhan** [comment]
@dpeng817 my understanding is our infra isn't schematized. but to double confirm that this column addition wont break our existing infra.

**Comment by dpeng817** [comment]
@yuhan you will need to have a change to the `telemetry_to_snowflake` job that adds a column to the snowflake table, and explicitly extracts this new column out of the payload. But no change is needed in the actual infrastructure.

File: python_modules/dagster/dagster/_core/telemetry.py [hunk 1]
```diff
@@ -133,6 +144,19 @@ def get_python_version():
     return "{}.{}.{}".format(version.major, version.minor, version.micro)
 
 
+def get_is_known_ci_env():
+    # Many CI tools will use `CI` key which lets us know for sure it's a CI env
+    if os.environ.get("CI") is True:
```
Comment (dpeng817) [inline python_modules/dagster/dagster/_core/telemetry.py:149]: i slight worry about false positives on this but eh

**Review by dpeng817 — APPROVED** [review]

File: python_modules/dagster/dagster/_core/telemetry.py (hunk 1 above)
Comment (yuhan) [reply python_modules/dagster/dagster/_core/telemetry.py:149]: im actually pretty confident here as 1) if you set a "CI" env var that says CI = true, i think it's safe to say it's a CI. 2) most of the links that i put next to the KNOWN_CI_ENV_VAR_KEYS has `CI = true` [plus this request](https://issues.jenkins.io/browse/JENKINS-36707) in jenkins, which makes me think it's a common practice.

**Pull request merged**